### PR TITLE
Select: Add disabled + disabled options + groups

### LINKED ISSUE #1062: Add disabled option to Select
We are missing a way to disabled options inside the Select component. This is a native feature to HTML.

### PR DESCRIPTION
**Pull request by francocorreasosa**
Extends #1064 
Resolves #1062

### CODE REVIEW

File: core/components/atoms/select/select.js [hunk 1]
```diff
@@ -20,14 +22,14 @@ const Select = ({ options, ...props }) => {
...
       {/* First option will be selected if there is no value passed as a prop */}
-      <option disabled hidden selected={!props.value} value="" {...Automation('select.option')}>
+      <StyledOption disabled hidden selected={!props.value} value="" {...Automation('select.option')}>
         {props.placeholder}
-      </option>
+      </StyledOption>
 
       {options.map((option, index) => (
-        <option key={index} value={option.value} {...Automation('select.option')}>
+        <StyledOption key={index} value={option.value} readOnly={option.readOnly} disabled={option.readOnly} {...Automation('select.option')}>
```
Comment (siddharthkp) [inline core/components/atoms/select/select.js]: Err why?

Comment (rakhi2104) [reply core/components/atoms/select/select.js]: the readOnly styling can be implemented this way. Tried css selectors, but didn't work.

Comment (francocorreasosa) [reply core/components/atoms/select/select.js]: We can just use the `disabled` attribute of the `<select />`

Comment (rakhi2104) [reply core/components/atoms/select/select.js]: [here](https://github.com/auth0/cosmos/issues/1062#issuecomment-435567902) I discussed the hardly noticeable difference of the disabled select option

Comment (siddharthkp) [reply core/components/atoms/select/select.js]: We need to use the disabled attribute either way for accessibility, I think we can add our custom styles to make it look more disabled 🙂

Comment (francocorreasosa) [reply core/components/atoms/select/select.js]: Fixed it :)

**Comment by francocorreasosa** [comment]
@landitus can you take a look whenever you have time?

**Comment by landitus** [comment]
I'm attaching the states for reference

screen shot 2018-11-06 at 10 53 19

**Comment by francocorreasosa** [comment]
Fixed, and added reference to the W3C in the docs :)

**Comment by landitus** [comment]
Let's make the read-only opacity 1, so it's just a native read-only select

**Comment by landitus** [comment]
Sorry about the confusion @francocorreasosa, as Selects don't have readonly, only disabled. 

Here's the updated design. The disabled opacity is 0.5.

**Comment by francocorreasosa** [comment]
No problem @landitus. Just updated the code to reflect that, thanks.

**Pull request merged**